tutorials/migration_light_api.md

```diff
@@ -0,0 +1,144 @@
+
+### Properties
+
+Most of Gazebo-classic's Light API is related to setting
+and getting properties. These functions are great candidates to have
+equivalents on Gazebo because the Entity-Component-System architecture is
+perfect for setting components (properties) into entities such as lights.
+This section focuses on migrating from APIs provided through the
+`gazebo::rendering::Light` class.
+
+---
+
+Classic | Gazebo
+-- | --
+CastShadows | `ignition::gazebo::Light::CastShadows`
+Clone | TODO
+DiffuseColor | `ignition::gazebo::Light::DiffuseColor`
+Direction | `ignition::gazebo::Light::Direction`
+FillMsg | TODO
+Id | `ignition::gazebo::Light::Entity`
+LightType | `ignition::gazebo::Light::Type`
+Name | `ignition::gazebo::Light::Name`
+Position | `ignition::gazebo::Light::Pose`
+Rotation | `ignition::gazebo::Light::Pose``
+SetAttenuation | use `ignition::gazebo::Light::SetAttenuation\*`
+SetCastShadows | `ignition::gazebo::Light::SetCastShadows`
+SetDiffuseColor | `ignition::gazebo::Light::SetDiffuseColor`
+SetDirection | `ignition::gazebo::Light::SetDirection`
+SetLightType | TODO
+SetName | TODO
+SetPosition | TODO
+SetRange | `ignition::gazebo::Light::SetAttenuationRange`
+SetRotation | TODO
+SetSelected |  Selection is client-specific, not porting
+SetSpecularColor | `ignition::gazebo::Light::SetSpecularColor`
+SetSpotFalloff | `ignition::gazebo::Light::SetSpotFalloff`
+SetSpotInnerAngle | ignition::gazebo::Light::SetSpotInnerAngle`
+SetSpotOuterAngle | ignition::gazebo::Light::SetSpotOuterAngle`
+SetVisible | TODO
+ShowVisual | TODO
+SpecularColor | ignition::gazebo::Light::SetSpecularColor`
+ToggleShowVisual | TODO
+Type | `ignition::gazebo::Light::Type`
+Visible | TODO
+WorldPose | TODO
+---
+
+
+## Read family
+
+These APIs deal with reading information related to child / parent
+relationships. As mentioned earlier, these APIs in Gazebo
+classic are provided by the `gazebo::physics::Light` class. Most of the
+APIs are inherited from the base `gazebo::physics::Entity` class in classic.
+We will only list the relevant ones here.
+
+The main difference in these APIs across Gazebo generations is that
+on classic, they deal with shared pointers to entities, while on Gazebo,
+they deal with entity IDs.
+
+---
+
+Classic | Gazebo
+-- | --
+GetParent | `ignition::gazebo::Light::Parent`
+GetParentId | `ignition::gazebo::Light::Parent`
+GetWorld | `ignition::gazebo::worldEntity`
+
+---
+
+## Write family
+
+These functions deal with modifying the entity tree, attaching children to new
+parents. As mentioned earlier, these APIs in Gazebo classic are provided by the
+`gazebo::physics::Light` class. Most of the APIs are inherited from the base
+`gazebo::physics::Entity` class in classic. We will only list the relevant ones
+here.
+
+---
+
+Classic | Gazebo
+-- | --
+AddChild | Not supported
+RemoveChild | Not supported
+RemoveChildren | Not supported
+SetParent | TODO
+SetWorld | TODO
+
```

Comment:
I assume that is currently no possible to use a write-like method in Gazebo, is this correct? If so, we can mention it.